app/controllers/application_controller.rb

```diff
@@ -5,8 +5,8 @@ class ApplicationController < ActionController::Base
 
   attr_reader :current_user, :auth_token
   before_action :set_user
-  after_action :verify_authorized, except: :index
-  after_action :verify_policy_scoped, only: :index
+  after_action :verify_authorized, except: [:index, :export_group]
```

Comment:
If you're excluding the export_group action from needing to verify that a single transcription was authorized, you need to instead verify that the the collection was scoped, so `:export_group` should also be added to the next line.

Reply:
Right, I tried that when I initially set it up, but because I need to use the editor scope (scope to transcriptions that the user can edit) which looks like this - `Scope.new(current_user, Transcription).editor_policy_scope` - rather than the standard `policy_scope` method (which scopes to transcriptions that the user can view), it doesn't pass the `verify_policy_scoped` check. Do you think it's worth trying to set up a check to make sure that the `editor_policy_scope` was applied? I'm guessing it _should_ be possible.

Reply:
`verify_authorized` is for verifying that member routes authorized the record they're returning. `verify_policy_scoped` is for verifying that collection routes authorized the collection they're returning via a policy scope. 

So you want to add `export_group` to the `verify_policy_scoped` action, making sure to not also check it with `verify_authorized`.

Reply:
Right right, so that didn't work out of the box. I set up the namespacing as described in the link you included and it now works when I add `:export_group` to the `:verify_policy_scoped` list. 

Let me know what you think of how I've got it set up now @zwolf

Reply:
Very interesting--gotta admit, the repetition is an issue and needs to be DRYed up. You're declaring TranscriptionPolicy (and its `::Scope`) twice along with the `editor_policy_scope` method in there. I don't think you could ever use the first one, as it would be overridden below. 

But I don't know exactly what the Pundit-approved answer to this would be. This isn't quite it, but it's def close and it does seem to be doing mostly what you expect, though maybe not precisely _how_ you expect. It's just pretty hard to parse what's going on in that policy for the non-export policy queries. Maybe diving into the Pundit source would help clear up how `policy_scope` is using the args you give it, so that you can point it right to the method you want it to use?

Reply:
Ah yeah, I forgot to take out the `editor_policy_scope` method out of the regular `TranscriptionPolicy` class when I first committed the changes, that's removed now.

I took a look at the source code, it looks like pundit expects there to be only one policy_scope for each object type (e.g. one policy scope for a `Transcription` object), and if there are additional policy scopes for an object type, they must be declared by creating a separate class for each scope.

The only arguments that `policy_scope` takes are `user` and `object` (see [policy_scope](https://github.com/varvet/pundit/blob/6be4621538f916124f76b57460c8cc46597a3b6a/lib/pundit.rb#L85) method), and the correct method to execute is deduced via the type of `object`, so it doesn't look like there's a way to pass any params to tell pundit to use one method over another for policy_scope, if that makes sense.

I'm not seeing any way to rework this to make it do what I need it to do in a better way, maybe lets talk and see if we might be able to come up with something together?

Reply:
tbh I don’t love the idea of leaving code in there that works, but we don’t know exactly why it works, even though the tests pass. The verification methods have got to run so that we’re certain everything’s locked down. This isn’t an overly complex use case for Pundit, but I’m not certain we’re using it correctly here. Worth further investigation, imo.

Reply:
> tbh I don’t love the idea of leaving code in there that works, but we don’t know exactly why it works, even though the tests pass. 

It makes sense why it works, it's just a matter of it being a clunky way of executing the desired behavior. 

Did some more digging, it looks like there's an [open issue](https://github.com/varvet/pundit/issues/368) for getting a scope for a specific action in pundit, but there appears to be no accepted/recommended way of doing it that also passes the `:verify_policy_scoped` check. [This comment](https://github.com/varvet/pundit/issues/368#issuecomment-196111115) specifically lays out a way of setting up method-specific scopes, which I like, but it doesn't work with `verify_policy_scoped` because the `policy_scope` method isn't invoked directly.

Based on [this comment](https://github.com/varvet/pundit/issues/368#issuecomment-196215250), It looks to me like one of the original designers of the interface was aware of this shortcoming, but didn't see method-specific scopes as an essential feature for the gem.

Reply:
That's a series of exasperating reads. There's a few solutions in there, my primary concern is ensuring that we know the policy is being used when a request is made. I saw your request to talk offline, will respond soon.

Reply:
Code updates to authorize `export_group` based on workflow are done :smiley: